src/transformers/models/bartpho/tokenization_bartpho.py

```diff
@@ -0,0 +1,302 @@
+# coding=utf-8
+# Copyright 2021 VinAI Research and the HuggingFace Inc. team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License
+""" Tokenization classes for BARTpho-syllable model."""
+
+
+import os
+from shutil import copyfile
+from typing import Any, Dict, List, Optional, Tuple
+
+import sentencepiece as spm
+
+from ...tokenization_utils import AddedToken, PreTrainedTokenizer
+from ...utils import logging
+
+
+logger = logging.get_logger(__name__)
+
+SPIECE_UNDERLINE = "▁"
+
+VOCAB_FILES_NAMES = {"vocab_file": "sentencepiece.bpe.model", "reduced_vocab_file": "dict.txt"}
+
+PRETRAINED_VOCAB_FILES_MAP = {
+    "vocab_file": {
+        "vinai/bartpho-syllable": "https://huggingface.co/vinai/bartpho-syllable/resolve/main/sentencepiece.bpe.model",
+    },
+    "reduced_vocab_file": {
+        "vinai/bartpho-syllable": "https://huggingface.co/vinai/bartpho-syllable/resolve/main/dict.txt",
+    },
+}
+
+PRETRAINED_POSITIONAL_EMBEDDINGS_SIZES = {"vinai/bartpho-syllable": 1024}
+
+
+class BartphoTokenizer(PreTrainedTokenizer):
+    """
+    Adapted from :class:`~transformers.XLMRobertaTokenizer`. Based on `SentencePiece
+    <https://github.com/google/sentencepiece>`__.
+
+    This tokenizer inherits from :class:`~transformers.PreTrainedTokenizer` which contains most of the main methods.
+    Users should refer to this superclass for more information regarding those methods.
+
+    Args:
+        vocab_file (:obj:`str`):
+            Path to the vocabulary file.
+        reduced_vocab_file (:obj:`str`):
+            Path to the reduced vocabulary file.
```

Comment:
Could you elaborate on the difference with the `vocab_file`?

Reply:
I did not train a sentencepiece for Vietnamese.
BARTpho-syllable employs the existing pre-trained "sentencepiece" model from XLMRoBERTaTokenizer, and this pre-trained model is referred to as a vocab of 250K types.
reduced_vocab_file is a vocab containing 40 Vietnamese-focused types extracted from the XLMRoBERTaTokenizer vocab of 250K types.
Other languages can thus simply reuse BartphoTokenizer with their own reduced_vocab_file.

Reply:
It would be great if you could add 1 or 2 sentences directly to the docstring :-)

Reply:
I join the previous comments, I think it would be really nice to have some more details about `reduced_vocab_file` in this docstring.